### PR TITLE
Clarify IBM PEF attestation procedure

### PR DESCRIPTION
**Pull request by gdhh**
These corrections were suggested by Ram and I.

### CODE REVIEW

**Comment by MikeCamel** [comment]
Thank you _very_ much for this, and apologies for the lack of response so far.  We are stuck in a spree of demo deadline coding and documentation, but will certainly return to this once that's over.  I've already started looking at it, but don't have enough detail to comment yet.  Other eyes will join mine soon.

File: docs/attestation/ibm/pef/process.msc [hunk 1]
```diff
@@ -2,34 +2,83 @@
 msc {
   tenant     [textbgcolor="green"],
   host       [textbgcolor="red"],
-  bios       [textbgcolor="orange"],
+  SLOF       [textbgcolor="orange"],
```
Comment (connorkuehl) [inline docs/attestation/ibm/pef/process.msc:5]: What's SLOF?

**Comment by gdhh** [comment]
Connor Thanks for your response. I will be happy to supply the requested 
text document.  However, I have limited experience with github so I will 
need some guidance.

  * What text editor or formatting system do you want the document
    written in?
  * Why do you have the document covering all architectures under:
    https://github.com/enarx/enarx/tree/master/docs/attestation#sev
    which appears to only apply to AMD SEV. Why would anyone look for
    information on IBM or Intel SEV's in this document
      o Is the editing system similar to Latex where there is a master
        that includes information from sub folders?
      o If so what sub folder should I be working in?

These questions should give you an indication of how little I know. 
Point me in the correct direction and hopefully I can catch up quickly.

Thanks,

Guerney

.

File: docs/attestation/ibm/pef/process.msc (hunk 1 above)
[the remark this answers is not in the document]
Comment (gdhh) [reply docs/attestation/ibm/pef/process.msc:5]: SLOF is the actual name of the firmware that is roughly equivalent to BIOS in Power. It stands for Simline Open [Firmware](https://openbios.info/SLOF)

File: docs/attestation/ibm/pef/process.msc [hunk 2]
```diff
@@ -2,34 +2,83 @@
...
+user of a cloud infrastructure. The flow does not specify where the trusted component is located.
+In particular, this flow says the infrastructure provider must give the identity of the target 
+machine to something trusted by the end user. This flow also does not specify how the VM is distributed
+to the cloud provider.\n\n For PEF, attestation can only occur when the target machine is known.
+The required certificate chain flows are documented elsewhere. If the lock box is not created 
+and inserted when the VM is initially generated, then the lockbox has to be created
+and inserted into the image prior to execution on the target machine. The normal PEF flows
+are illustrated after VM START. It is important to note that if the seed is ever in an untrusted
+or insecure location, the owner of the VM must assume the security is compromised."];
+
```
[the remark this answers is not in the document]
Comment (gdhh) [reply docs/attestation/ibm/pef/process.msc:27]: I agree that the back ground information will be more consumable as prose. As I am somewhat new to github, I do not know what tools you would like me to use to produce this prose. Please point to a tutorial. Thanks

File: docs/attestation/ibm/pef/process.msc [hunk 3]
```diff
@@ -2,34 +2,83 @@
...
+  tpm box tpm [label="decrypt lockbox(private-key)"];
+  tpm => ultravisor [label="seed"];
+  ultravisor box ultravisor [label="gen symmetric key"];
+  ultravisor box ultravisor [label="gen HMAC key"];
+  ultravisor box ultravisor [label="check payload integrity(HAMC key)"];
+  ultravisor=>host [linecolor=red,label="terminate if integrity check fails"];
+  ultravisor box ultravisor [label="extract payload from esm operand"];
+  ultravisor box ultravisor [label="decrypt\n(symmetric key,payload)"];
+  ultravisor box ultravisor [label="validate VM integrity"];
+  ultravisor=>host [linecolor=red,label="termintate if integrity check fails"];
```
[the remark this answers is not in the document]
Comment (gdhh) [reply docs/attestation/ibm/pef/process.msc:73]: agree

File: docs/attestation/ibm/pef/process.msc [hunk 4]
```diff
@@ -2,34 +2,83 @@
...
+  kernel box kernel [label="ESM instruction"];
+  kernel=>ultravisor [label="deliver esmoperand"];
+  ultravisor box ultravisor [label="extract lockbox from esmoperand"];
+  ultravisor=>host [linecolor=red, label="terminate if valid lock box not found"];
+  ultravisor => tpm [label="deliver lockbox"];
+  tpm box tpm [label="decrypt lockbox(private-key)"];
+  tpm => ultravisor [label="seed"];
+  ultravisor box ultravisor [label="gen symmetric key"];
+  ultravisor box ultravisor [label="gen HMAC key"];
+  ultravisor box ultravisor [label="check payload integrity(HAMC key)"];
```
[the remark this answers is not in the document]
Comment (gdhh) [reply docs/attestation/ibm/pef/process.msc:68]: agree

**Comment by lkatalin** [comment]
> Connor Thanks for your response. I will be happy to supply the requested text document.  However, I have limited experience with github so I will need some guidance. * What text editor or formatting system do you want the document written in? * Why do you have the document covering all architectures under: https://github.com/enarx/enarx/tree/master/docs/attestation#sev which appears to only apply to AMD SEV. Why would anyone look for information on IBM or Intel SEV's in this document o Is the editing system similar to Latex where there is a master that includes information from sub folders? o If so what sub folder should I be working in? These questions should give you an indication of how little I know. Point me in the correct direction and hopefully I can catch up quickly. Thanks, Guerney .

@gdhh Thanks for following up on our comments. I can try to answer some of your questions. 

Re: Which format for the text document, we use markdown and that is the format you see on the [example pages](https://github.com/enarx/enarx/blob/master/docs/attestation/README.md). There is a good tutorial [here](https://docs.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax).

Re: The organization of our documentation, I'm not sure if it has changed since you asked about it, but we have an overview of all the architectures located [here](https://github.com/enarx/enarx/blob/master/docs/attestation/README.md) and this includes links for the specific technologies within the overview (still a work in progress). The sub-folders contain the diagrams related to each technology; so in that sense, yes, it is similar to Latex because you can use those diagrams by referring to them from the main overview document. If you check the overview doc in a text editor you'll see the references (they render as images when you view from the browser).

The relevant docs and folders for this PR would be the main [overview doc](https://github.com/enarx/enarx/blob/master/docs/attestation/README.md) and/or the imb/pef [subfolder](https://github.com/enarx/enarx/tree/master/docs/attestation/ibm/pef) for diagrams. I believe the requested accompanying text would go in the main overview doc.

@connorkuehl Let me know if any of that is incorrect as I wasn't around for the context of this PR initially.

**Comment by connorkuehl** [comment]
> @gdhh Thanks for following up on our comments. I can try to answer some of your questions.
> 
> Re: Which format for the text document, we use markdown and that is the format you see on the [example pages](https://github.com/enarx/enarx/blob/master/docs/attestation/README.md). There is a good tutorial [here](https://docs.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax).
> 
> Re: The organization of our documentation, I'm not sure if it has changed since you asked about it, but we have an overview of all the architectures located [here](https://github.com/enarx/enarx/blob/master/docs/attestation/README.md) and this includes links for the specific technologies within the overview (still a work in progress). The sub-folders contain the diagrams related to each technology; so in that sense, yes, it is similar to Latex because you can use those diagrams by referring to them from the main overview document. If you check the overview doc in a text editor you'll see the references (they render as images when you view from the browser).
> 
> The relevant docs and folders for this PR would be the main [overview doc](https://github.com/enarx/enarx/blob/master/docs/attestation/README.md) and/or the imb/pef [subfolder](https://github.com/enarx/enarx/tree/master/docs/attestation/ibm/pef) for diagrams. I believe the requested accompanying text would go in the main overview doc.
> 
> @connorkuehl Let me know if any of that is incorrect as I wasn't around for the context of this PR initially.

@lkatalin this all looks good to me.

@gdhh If you're still interested in pursuing this pull request, my offer to collaborate in my email on June 4th still stands.

**Pull request merged**